For Editorial review (#1226)
mulesoft/docs-connectors: x12-edi/2.5/modules/ROOT/pages/x12-edi-connector-config-topics.adoc

= X12 EDI Connector - Schema and Trading Partner Configuration - Mule 4

Use these additional XML configuration options to:

> 

Reviewer: Lines 5 through 31 hidden, unedited.

</x12-edi:config>
----

> If you don't include any schemas in the configuration, X12 EDI Connector attempts to load an appropriate schema at runtime for each transaction set by using the standard X12 schema definitions provided with the connector. However, in this case you won't see any metadata to define the structure of the

Reviewer: ```suggestion
If you don't include any schemas in the configuration, X12 EDI Connector attempts to load the standard X12 schema definitions provided with the connector. However, in this case you won't see any metadata to define the structure of the
```

data passed in and out of the connector.

If the schema definition configuration includes a `997` or `999` Functional Acknowledgment schema, then that schema is used to generate Functional Acknowledgments.

> NOTE: To use the Functional Acknowledgment schema type (`997` or `999`) it must match the `generate999Acks` Parser parameter (`false` for `997`, `true` for `999`).

Reviewer: ```suggestion
NOTE: To use the Functional Acknowledgment schema type (`997` or `999`), it must match the `generate999Acks` Parser parameter (`false` for `997`, `true` for `999`).
```
What is "it"?

|XML Value |Studio Option
|formValidation="X12"/"HIPAA_SNIP1"/"HIPAA_SNIP2" |The schema form and validation level can be either X12 or HIPAA. The default is X12.
|schemas=list of values |List of schema paths the connector can use. These paths can be either a file system or `classpath`.

> |characterEncoding="PLATFORM"/"ASCII"/"ISO8859_1"/"IBM1047"/"UTF8" |The character encoding used for both send and receive messages. The default `PLATFORM` is the Mule platform default encoding.

Reviewer: What is "the Mule platform"? We use the abbreviation "Mule" to represent Mule runtime engine.

> |stringCharacterSet="BASIC"/"EXTENDED"/"UNRESTRICTED" |Use a substitution character replacement for invalid characters, or reject them as errors if set to `none` or enabled for receive messages in the parser options.  The default is `EXTENDED` for string data.

Reviewer: Isn't "substitution character replacement" redundant?

Reviewer: "Either use a...characters or reject...."

|stringSubstitutionChar="?" |Substitution character used to replace invalid characters in string values. The default is `none`.

> |versionIdentifierSuffix="XYZ" |Version identifier code suffix for the group (GS). Setting this value affects both Send and Receive message handling. On the receive side, the GS08 version code must contain X12 schema version (for example, `004010`, `005010`, and so on) followed by this suffix. Not setting this value means the receive-only just verifies that the GS version identifier code starts with the configured schema version. On the Send side, the combination of X12 schema version and this suffix comprises the GS08 value.

Reviewer: Note that earlier text lowercases send and receive....
In fact, this paragraph mixes the formats.
Standardize.

Reviewer: On the receive side, the GS08 version code must contain X12 schema version (for example, `004010`, `005010`, and so on) followed by this suffix.
>
On the receive side, the GS08 version code must contain **the** X12 schema version (for example, `004010`, `005010`, and so on)**,** followed by this suffix.

Reviewer: Lines 51-56 hidden

[%header%autowidth.spread]
|===
|XML Value |Studio Option

> |interchangeIdQualifierSelf="ZZ" |The ID qualifier for Mule in Exchange (known as the ISA segment). Use this value in combination with the `Interchange ID` value. Both must be either set or unset.

Reviewer: ```suggestion
|interchangeIdQualifierSelf="ZZ" |The ID qualifier for Mule runtime engine in Exchange (known as the _ISA segment_). Use this value in combination with the `Interchange ID` value. Both values must be either specified or not specified.
```

> |interchangeIdSelf="MULE" |The ID qualifier for Mule in Exchange (known as the ISA segment). Use this value in combination with the `Interchange ID` value. Both must be either set or unset.

Reviewer: ditto prev. edit to same text

> |groupIdSelf="MULEAPP" |Application Code for Mule in a group (GS segment).

Reviewer: ```suggestion
|groupIdSelf="MULEAPP" |Application code for Mule runtime engine in a group (GS segment).
```

[%header%autowidth.spread]
|===
|XML Value |Studio Option
|interchangeIdQualifierSelf="ZZ" |The ID qualifier for a partner in Exchange (known as the ISA segment). Use this value in combination with the `Interchange ID` value. Both must be either set or unset.

> |interchangeIdSelf="PARTNER" |The ID qualifier for a partner in Exchange (known as the ISA segment). Use this value in combination with the `Interchange ID` value. Both must be either set or unset.

Reviewer: ditto prev. edits to same text

> |groupIdSelf="PARTNERAPP" |Application Code for a partner in a group (GS segment)

Reviewer: ditto
Add period.

|===

> 

Reviewer: Lines 77-87 hidden

[%header%autowidth.spread]
|===
|XML Value (default) |Studio Option
|enforceLengthLimits="true" |Enforce the minimum and maximum lengths for receive values.
|enforceCharacterSet="true" |Enforce the allowed character set.
|enforceValueRepeats="true" |Enforce the repetition count limits for receive values.
|allowUnknownSegments="false" |Allow unknown segments in a transaction set.
|enforceSegmentOrder="true" |Enforce the segment order in a transaction set.
|allowUnusedSegments="false" |Allow segments marked as `Unused` in a transaction set.
|enforceSegmentRepeats="true" |Enforce segment repetition count limits in a transaction set.

> |requireUniqueInterchanges="true" |Require unique ISA Interchange Control Numbers (such as ISA13). The default behavior records the interchange numbers previously processed and rejects duplicate interchange numbers from the same partner. It is derived from the interchange Sender and Receiver identification. Setting to `false` allows received interchange processing to continue and relies on the application flow to track the numbers and handle them appropriately.

Reviewer 2: ```suggestion
|requireUniqueInterchanges="true" |Require unique ISA Interchange Control Numbers (such as ISA13). The default behavior records the interchange numbers previously processed and rejects duplicate interchange numbers from the same partner. It is derived from the interchange Sender and Receiver identification. Setting to `false` enables received interchange processing to continue and relies on the application flow to track the numbers and handle them appropriately.
```

|daysToStore="30" |Minimum number of days to store interchange, group, and transaction set numbers for uniqueness checking.
|ackAllSets="false" |Include a separate AK2/AK5 (`997`) or AK2/IK5 (`999`) acknowledgment for every Received transaction set. By default (`false`), transaction sets containing errors are part of the acknowledgment, with all other transaction sets implicitly acknowledged. Changing to `true` acknowledges each Received transaction set.
|generate999Acks="false" |Generate `999` Implementation Acknowledgments instead of `997` Functional Acknowledgments. Setting to `false`, the `997` Functional Acknowledgment transaction sets are generated for each received interchange. Setting to `true` generates `999` Implementation Acknowledgments instead. Support for `999` Implementation Acknowledgments does not include CTX segment generation.
|reportSegmentErrors="true" |Report segment error details to the sender in the `997`or`999` flag.
Setting to `true` includes the details of any segment errors that are generated in the `997`or`999` flag. `false` does not include the details.

> |includeFASchema="true" |Expect `997` or `999` Functional Acknowledgments and include the `997` or `999` schema. `true` automatically includes the schema for the `997` or `999` acknowledgment transaction sets within the set of schemas used by X12 EDI connector. Setting to false, means directly specifying the `997` or `999` schema to process as the input. The schemas used for generating `997` or `999` Functional Acknowledgments are hardcoded and cannot be modified.

Reviewer 2: ```suggestion
|includeFASchema="true" |Expect `997` or `999` Functional Acknowledgments and include the `997` or `999` schema. `true` automatically includes the schema for the `997` or `999` acknowledgment transaction sets within the set of schemas used by X12 EDI connector. Setting to `false`, means directly specifying the `997` or `999` schema to process as the input. The schemas used for generating `997` or `999` Functional Acknowledgments are hardcoded and cannot be modified.
```

|enforceSegmentOrder="true" |Enforce the segment order in a transaction set
|allowUnusedSegments="false" |Allow segments marked as `Unused` in a transaction set

> |enforceSegmentRepeats="true" |Enforce segment repetition count limits in a transaction set

Reviewer: periods

Reviewer: RESUME
Part 3 of CE

> |requireUniqueInterchanges="true" |Require unique ISA Interchange Control Numbers (such as ISA13). The default behavior records the interchange numbers previously processed and rejects duplicate interchange numbers from the same partner. It is derived by the interchange Sender and Receiver identification. Set to `false` allows received interchange processing to continue. This relies on the application flow to track the numbers and handle them appropriately.

Reviewer: "It is derived..." : What is "it"?
"derived by <doing something>" or do you mean "derived from" the IDs?

Reviewer: Why do we describe "Set to `false`" when the XML value we are describing is the `true` version?

Reviewer: In the final sentence, to what does "This" refer?

> |requireUniqueGroups="false" |Enforce globally unique Group Control Numbers (GS06) for received functional groups. By default, group numbers need to be unique within an interchange. Set to `True` requires group numbers to be unique across all interchanges received from the same partner and application. This is derived by the interchange sender and receiver identification combined with the functional group sender and receiver application codes

Reviewer: This XML value is "false" so why do we define it as enforcing globally unique numbers (which would be "true", right)?

Why describe set to `True` when we are describing the "false" value?

Note also that while this text init. caps. "Set to `True`," previous text lowercases "Set to `false`". Standardize throughout; be consistent to avoid audience confusion.

Reviewer: ditto prev. comments about "It (This) is derived by...."

Reviewer: End with period.

> |requireUniqueTransactionSets="false" |Enforce globally unique Transaction Set Control Numbers (ST02) for received transaction sets. By default, transaction set control numbers only need to be unique with a particular functional group. Set to `True` requires transaction set numbers to be unique across all functional groups received from the same partner and application.  This is derived by the interchange sender and receiver identification, combined with the functional group sender and receiver application codes.

Reviewer: "numbers only need to be unique with" > "numbers need to be unique only within"

Reviewer: Ditto previous edits about describing both true and false settings and inconsistent capitalization: GLOBAL for these sentences throughout.

Reviewer: Ditto "This is derived by" throughout; GLOBAL.

> |daysToStore="30" |Minimum number of days to store interchange, group, and transaction set numbers for uniqueness checking.

Reviewer: "uniqueness checking"? Is that a real term? lol

> |ackAllSets="false" |Include a separate AK2/AK5 (`997`) or AK2/IK5 (`999`) acknowledgment for every Received transaction set. By default (`false`), transaction sets containing errors are part of the acknowledgment, with all other transaction sets implicitly acknowledged. Changing to `true` acknowledges each Received transaction set.

Reviewer: So the value in the first column is always the default: is that why we describe both true and false settings in each definition?
If so, we should clarify that in the column heading.

> |generate999Acks="false" |Generate `999` Implementation Acknowledgments instead of `997` Functional Acknowledgments. Set to `false` the `997` Functional Acknowledgment transaction sets generate for each received interchange. Set to `true` generates `999` Implementation Acknowledgments instead. Support for `999` Implementation Acknowledgments does not include CTX segment generation.

Reviewer: Reword this to clarify it:
"Set to `false` the `997` Functional Acknowledgment transaction sets generate for each received interchange."
Comma after `false`? And are you saying that the sets "generate" instead of "are generated"?
Consider making it parallel with the sentence that follows it.

|reportSegmentErrors="true" |Report segment error details to the sender in the `997`or`999` flag.
Setting to `True` includes the details of any segment errors that are generated in the `997`or`999` flag. `False` does not include the details.

> |includeFASchema="true" |Expect `997` or `999` Functional Acknowledgments and include the `997` or `999` schema. `True` automatically includes the schema for the `997` or `999` acknowledgment transaction sets into the set of schemas used by X12 EDI connector. If false, you must directly specify the `997` or `999` schema to process these as the input. The schemas used for generating `997` or `999` Functional Acknowledgments are hardcoded and cannot be modified.

Reviewer: "...includes the schema...into the set of schemas..." is improper grammar. You do not "include into". Reword.

Reviewer: Modification error much like previous: "If false, you must...." ("you" are not false). Reword.

[%header%autowidth.spread]
|===
|XML Value |Studio Option
|dataSeparator="*" |Data element separator character. The default uses the configured value for all output messages and can be overridden at the message level.
|componentSeparator=">" |Component separator character. The default uses the configured value for all output messages and can be overridden at the message level.
|repetitionSeparator="U" |Repetition separator character. The default uses the configured value for all output messages and can be overridden at the message level. `U` means repetitions are not used
|segmentTerminator="~" |Segment terminator character. The default uses the configured value for all output messages and can be overridden at the message level.

> |lineEnding="NONE"/"LF"/"CRLF"/"CR" |Line ending to add between segments, the default is `NONE`. You can add line endings between segments to improve message text output readability.

Reviewer: ```suggestion
|lineEnding="NONE"/"LF"/"CRLF"/"CR" |Line ending to add between segments. The default is `NONE`. You can add line endings between segments to improve message text output readability.
```

> |sendUniqueGroupNumbers="false" |Send unique Group Control Numbers. `False` (default) assigns functional group control numbers sequentially within each interchange and reuses them in different interchanges. `True` assigns unique group numbers across all interchanges sent to the same partner and application. It is derived by the interchange sender and receiver identification combined with the functional group sender and receiver application codes.

Reviewer: ditto "It is derived by"

> |sendUniqueTransactionNumbers="false" |Send unique Transaction Set Control Numbers. `False` (default) assigns transaction set control numbers sequentially within each functional group and reuses them in different groups. `True` assigns unique transaction set numbers across all interchanges sent to the same partner and application. It is derived by the interchange sender and receiver identification, combined with the functional group sender and receiver application codes.

Reviewer: ditto

> |implementationConventionReference="" |Implementation convention reference for transactions (ST segment). Setting this value uses it as the ST Implementation Convention Reference unless overridden in the message parameters.

Reviewer: "Setting this value uses it as..." is incorrectly worded: setting it does not use it as. Reword.

|initialInterchangeNumber="1" |The initial Interchange Control Number used for outgoing messages.
|initialGroupNumber="1" |The initial Group Control Number used for outgoing messages.
|initialSetNumber="1" |The initial Transaction Set Control Number used for outgoing messages.
|ackRequested="false" |Request acknowledgments for sent transactions flag. If true, `997` or `999` acknowledgments are requested for all sent transactions.
|defaultUsageIndicator="P" |Default ISA15 interchange usage indicator: `I` for Information, `P` for Production Data, `T` for Test Data.

> |useSuppliedValues="false" |Use values from supplied data for control segment identifiers (ISA/IEA, GS/GE, ST/SE segments). The default is always to generate control numbers when writing allowing you to use chosen values.

Reviewer: This is really unclear: the value here is false, but the description is true, to use values. And the default sounds like it is true as well, although there is at least one clarifying comma missing.
Reword the end to say "Setting this value so that a control number is always generated during a write operation enables you to"...use "chosen" values? Which chosen values? chosen by what/whom?

|outputEdiMimeType="APPLICATION_PLAIN"/"APPLICATION_EDIX12" |Output MIME type to be set for the message, either the default `application/plain` or X12-specific alternative `application/edi-x12`.

> |writeEnforceLengthLimits="true" |Enforce minimum and maximum lengths for write values. The default of `true` throws an exception when an element is too long or too short. `False` leaves the values as is

Reviewer: ```suggestion
|writeEnforceLengthLimits="true" |Enforce minimum and maximum lengths for write values. The default of `true` throws an exception when an element is too long or too short. `False` leaves the values as-is.
```
